### ISSUE TITLE
[test] Should test ROM also scramble the SRAM when initializing on FPGA / Verilator?

### ISSUE THREAD

**Issue by msfschaffner** [lowRISC/opentitan#16280]
I just noticed that when the test ROM initializes the main SRAM (happens only in Verilator sims and FPGA runs), the test ROM does not use request a new scrambling key:
https://github.com/lowRISC/opentitan/blob/3ba0b5a3bb02b01272f341fc4d399ac3214b31a4/sw/device/lib/testing/test_rom/test_rom_start.S#L214-L222

The real ROM however does that:
https://github.com/lowRISC/opentitan/blob/3ba0b5a3bb02b01272f341fc4d399ac3214b31a4/sw/device/silicon_creator/rom/rom_start.S#L329-L333

Should this be aligned, or do we keep the default key on purpose in DV?
This may already have been answered somewhere, but I just wanted to make sure this was a conscious decision.

CC @tjaychen @sriyerg @weicaiyang @timothytrippel @alphan

**Comment by tjaychen** [comment]
yeah for this stuff it would be best to align test_rom / real_rom as much as possible. 
I don't completely remember, but we might have taken it out to speed up boot? Because otherwise we need to wait for entropy_src to collect its seeds and all that exciting stuff. 

Was this by any chance removed when @matutem went through and did a bunch of rom speed up?

**Comment by msfschaffner** [comment]
> yeah for this stuff it would be best to align test_rom / real_rom as much as possible. I don't completely remember, but we might have taken it out to speed up boot? Because otherwise we need to wait for entropy_src to collect its seeds and all that exciting stuff.
> 
> Was this by any chance removed when @matutem went through and did a bunch of rom speed up?

Yeah but wasn't that mostly intended for sim_dv tests? Given that we skip this entirely and backdoor initialize in sim_dv, adding the scrambling key request back to `test_rom` would not impact sim_dv runtimes...

**Comment by tjaychen** [comment]
o do we skip it compoletely for sim_dv? we can't make rom skip this code right? I thought that's why we maybe did not do it in the assembly so it doesn't kick off an entire stalled process for key update.

**Comment by weicaiyang** [comment]
> Given that we skip this entirely and backdoor initialize in sim_dv, adding the scrambling key request back to test_rom would not impact sim_dv runtimes...

If we want to add key request back, make sure we update TB to use the right key for sram backdoor write.

**Comment by msfschaffner** [comment]
no, the real ROM can't skip this, only the test ROM can.

I believe its mostly the SRAM init that takes a long time - the key update should be a bit faster (although OTP needs to churn the digest module for a couple of 100 cycles). 

If we do what @weicaiyang says, we could technically request a key, but let the TB initialize its content to make it more speedy.

**Comment by tjaychen** [comment]
actually the key update could take awhile because you take entropy from the entropy complex right?
That process goes through seed collection from ast, and the ast by default spits out seeds really slowly unless you set certain parameters.

That's why I think maybe we removed this to speed things up.  If we don't do key updates, we also don't do entropy requests.

**Comment by timothytrippel** [comment]
sorry late to the discussion here, but yes it was removed in #9316 and subsequently updated in #11857.

**Comment by tjaychen** [comment]
aha! so we have come full circle.  Thanks @timothytrippel
